src/components/ShareTrajectoryButton/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { Button, Tooltip } from "antd";
+
+import { TOOLTIP_COLOR } from "../../constants";
+import {
+    NetworkedSimFile,
+    LocalSimFile,
+    isLocalFileInterface,
+} from "../../state/trajectory/types";
+import { Share } from "../Icons";
+import ShareTrajectoryModal from "../ShareTrajectoryModal";
+
+import styles from "./style.css";
+
+interface ShareTrajectoryButtonProps {
+    simulariumFile: LocalSimFile | NetworkedSimFile;
+}
+
+const ShareTrajectoryButton = ({
+    simulariumFile,
+}: ShareTrajectoryButtonProps): JSX.Element => {
+    const [isSharing, setIsSharing] = React.useState(false);
+
+    const isLocalFile = isLocalFileInterface(simulariumFile);
+
+    const handleShare = () => {
+        setIsSharing(!isSharing);
+    };
+
+    const handleTooltipOpen = () => {
+        if (isSharing || !simulariumFile.name) {
+            return false;
+        } else return;
+    };
```

Comment:
Should this return true in the else branch?

Could also probably compress this to a one-liner, i.e.
```
const handleTooltipOpen = () => {
   return !isSharing && simulariumFile.name
}
```

Reply:
Also, is there any reason to have this as a separate function instead of inline on line 42? Since line 56 does inline too.

Reply:
I like both your suggestions and wish they worked because they are cleaner, but the under-the-hood mechanics of this antd prop make it so that the one liner function, and passing the values directly like on 56, cause the tooltip to be open all the time, rather than just when hovering over the button. Would love to streamline, but need the behavior to still be responsive somehow. 
56 works I think because the disabling is not responsive, its meant to be always active.

Reply:
Messed around with this, you can also just remove the `open` prop altogether and the hovering behavior will still work! It looks like the `DownloadTrajectoryMenu` button next to this one in doesn't include the `open` prop.

Reply:
When I remove the open prop I do get the desired behavior re: hover, but I'm also able to see the tooltip in an empty viewer, and when the sharing modal is open (also stays open when hovering over the transparent overlay of the modal), which I don't think is the desired behavior. I can check in with UX to confirm.

Reply:
Well actually I can delete that second return statement and it works, as long as i leave the rest, and it's cleaner for sure, thanks!
```
    const handleTooltipOpen = () => {
        if (isSharing || !simulariumFile.name) return false;
    };
```